src/sqlfluff/rules/L009.py

```diff
@@ -95,11 +100,39 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             if file_len != pos:
                 return None
 
-        # We're going to make an edit because we're appending to the end and there's
-        # no segment after it to match on. Otherwise we would never get a match!
-        return LintResult(
-            anchor=context.segment,
-            fixes=[
-                LintFix("edit", context.segment, [context.segment, NewlineSegment()])
-            ],
-        )
+        # Include current segment for complete stack.
+        complete_stack = list(context.raw_stack)
+        complete_stack.append(context.segment)
```

Comment:
@tunetheweb mypy is throwing an error here as the first line implies `List[RawSegment]`
and then I append a `BaseSegment` in the following line. Given that `RawSegment` inherits `BaseSegment` and I'm only touching the inherited properties/methods I'll in-line annotate this `List[BaseSegment]`. Unless there's a raw_to_base (or vice versa) method I've missed.

Reply:
Paging @barrywhart and @alanmcruickshank who know much more about this than me!